### PR TITLE
Create recipe for quetz-sql-authenticator

### PR DESCRIPTION
**Pull request by simonbohnen**
`quetz-sql-authenticator` is a simple authenticator for the quetz package server which stores credentials in a SQL table.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/quetz-sql-authenticator/meta.yaml
Comment (janjagusch) [inline recipes/quetz-sql-authenticator/meta.yaml:51]: I'm happy to co-maintain this recipe.

File: recipes/quetz-sql-authenticator/meta.yaml
Comment (janjagusch) [inline recipes/quetz-sql-authenticator/meta.yaml:15]: why don't we build for windows? isn't this a simple `noarch` package?

Comment (simonbohnen) [reply recipes/quetz-sql-authenticator/meta.yaml:15]: Apparently quetz is not available on Windows, see [this run](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=659110&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=2fb5b0a7-737c-5f5c-8f3f-f6db6174bacf&l=359).

File: recipes/quetz-sql-authenticator/meta.yaml
Comment (janjagusch) [inline recipes/quetz-sql-authenticator/meta.yaml:23]:
```suggestion
build:
  noarch: python
  script: {{ PYTHON }} -m pip install . -vv
  number: 0

requirements:
  host:
    - python >=3.6
    - pip
```

**Comment by wolfv** [comment]
happy to be a maintainer here! :)

File: recipes/quetz-sql-authenticator/meta.yaml
Comment (janjagusch) [inline recipes/quetz-sql-authenticator/meta.yaml:23]:
```suggestion
    - quetz >=0.6.1
```

Comment (janjagusch) [reply recipes/quetz-sql-authenticator/meta.yaml:23]: 0.6.1 is now available on CF.

**Comment by simonbohnen** [comment]
@conda-forge/help-python This PR is ready for review! 😁

**Review by janjagusch — APPROVED** [review]

File: recipes/quetz-sql-authenticator/meta.yaml
Comment (dopplershift) [inline recipes/quetz-sql-authenticator/meta.yaml:38]: Are you aware this URL gives a 404?

Comment (simonbohnen) [reply recipes/quetz-sql-authenticator/meta.yaml:38]: Yes, we're gonna make it public soon 👍🏼 Is that required to create the feedstock?

Comment (dopplershift) [reply recipes/quetz-sql-authenticator/meta.yaml:38]: No, it's not required, it just generally makes it easier to review by being able to click through the repository rather than downloading the sdist from PyPI.

It's also not at all clear to me what the benefit is for having a private repo when the code is already up on PyPI.

**Review by dopplershift — APPROVED** [review]

File: recipes/quetz-sql-authenticator/meta.yaml
Comment (simonbohnen) [reply recipes/quetz-sql-authenticator/meta.yaml:38]: I don't have the rights to make it public, but it's in the works.

**Pull request merged**
